### PR TITLE
2nd Assignment by 202503728 (Yeryung Oh)

### PR DESCRIPTION
**Pull request by ttomaki**
# Assignment Submission

## Student Information
- **Student ID**: [Your Student ID]
- **Name**: [Your Full Name]
- **Assignment**: [Assignment Number]

## Changed Files

```
submissions
└── 202503728
    └── assignment2
        ├── README.md
        ├── problem1
        │   ├── README.md
        │   └── main.py
        ├── problem2
        │   ├── README.md
        │   └── textops
        │       ├── __init__.py
        │       ├── clean
        │       │   ├── __init__.py
        │       │   └── filters.py
        │       ├── main.py
        │       └── tokenize
        │           ├── __init__.py
        │           └── word.py
        ├── problem3
        │   ├── README.md
        │   └── main.py
        ├── problem4
        │   ├── README.md
        │   └── dsops
        │       ├── __init__.py
        │       ├── main.py
        │       ├── split
        │       │   ├── __init__.py
        │       │   └── train_test.py
        │       └── stats
        │           ├── __init__.py
        │           └── labels.py
        ├── problem5
        │   ├── README.md
        │   └── cachekit
        │       ├── __init__.py
        │       └── main.py
        └── problem6
            ├── README.md
            └── main.py
```

### Problem 1
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 2
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 3
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 4 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 5 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

### Problem 6 (Optional)
- [ ] Implemented required functionality
- [ ] All tests pass locally

## Testing
- [ ] I have run `python grader.py` locally
- [ ] All implemented problems pass their tests
- [ ] I have tested edge cases mentioned in the problem statements

## Additional Notes

---
**Note**: This PR will be automatically validated for:
- Correct directory structure
- Valid Python syntax
- PR title format: `Nth Assignment by {student_id} (Full Name)`
- File organization compliance

### CODE REVIEW

File: submissions/202503728/assignment2/problem3/main.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+def merge_freqs(maps: list[dict[str, int]]) -> dict[str, int]:
+    # TODO: 구현하세요 (선택사항)
+    result = {}
+    for b in maps:
+        for ket,value in b.items():
+            result[key]=result.get(key,0)+value
+    return result
+
+
```
Comment (202403635-min) [inline submissions/202503728/assignment2/problem3/main.py:33]: 전체적으로 잘 짜여진 코드라고 생각됩니다. 다만, 마지막줄의 pass는 해당 코드에서 아무런 기능을 하고 있지 않으므로 생략해도 될 것 같습니다. 이 pass는 예제코드에서 run_demo()가 주석처리되었기 때문에 임시로 세워둔 것이라고 생각됩니다. 따라서 run_demo()가 작동하는 지금은 필요없는 코드가 아닐까요? 이 부분 뿐만 아니라 다른 부분들도 pass 코드가 남아있는 모습이 많이 발견됩니다. 이 점을 제외하곤 아주 완성도 있는 코드인 것 같습니다.
